lib/puppet/provider/git_deploy_key/gitlab.rb

```diff
@@ -69,7 +69,7 @@ def exists?
     project_id = get_project_id
 
     sshkey_hash = Hash.new
-    url = "#{gms_server}/api/v3/projects/#{project_id}/keys"
+    url = "#{gms_server}/api/v4/projects/#{project_id}/keys"
```

Comment:
/api/v4/projects/#{project_id}/keys does not exist anymore, the correct path is /api/v4/projects/#{project_id}/deploy_keys now

Reply:
Cheers, thanks for the catch!

Reply:
No problem, dont't forget to change this in the whole file ;)